### PR TITLE
fix sq_quote_buf_pretty

### PR DESCRIPTION
**Pull request by garimasi514**
fix, tests: make sure sq_quote_buf_pretty doesn't inadvertently drop tokens that are empty and null, add tests for sq_quote_buf_pretty (quote.c) around some basic scenarios and to document the behavior of the method for null and empty strings

### CODE REVIEW

File: t/helper/test-quote.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#include "strbuf.h"
+#include "string.h"
+
+int cmd__quote_buf_pretty(int argc, const char **argv)
+{
+	struct strbuf buf_payload = STRBUF_INIT;
+	char nullString[2];
+	char empty[5] = {'\0'};
+	
+	if (strcmp(argv[1], "nullString") == 0){
```
Comment (derrickstolee) [inline t/helper/test-quote.c]: `if (!strcmp(argv[1], "nullString")) {`

(git style)

File: t/helper/test-quote.c [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		strbuf_addstr(&buf_payload, ".");
+		sq_quote_buf_pretty(&buf_payload, nullString);
+		printf("%s", buf_payload.buf);
+		printf(".\n");
+		return 0;
+	}
+	
+	if (strcmp(argv[1], "") == 0){
```
Comment (derrickstolee) [inline t/helper/test-quote.c]: Same here.

File: t/helper/test-quote.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#include "string.h"
+
+int cmd__quote_buf_pretty(int argc, const char **argv)
+{
+	struct strbuf buf_payload = STRBUF_INIT;
+	char nullString[2];
+	char empty[5] = {'\0'};
+	
+	if (strcmp(argv[1], "nullString") == 0){
+		strbuf_addstr(&buf_payload, ".");
```
Comment (derrickstolee) [inline t/helper/test-quote.c:13]: `strbuf_addch(&buf_payload, '.');`

File: t/helper/test-quote.c [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+		sq_quote_buf_pretty(&buf_payload, nullString);
+		printf("%s", buf_payload.buf);
```
Comment (derrickstolee) [inline t/helper/test-quote.c]: `printf("%s.\n", buf_payload.buf);`

**Review by derrickstolee — CHANGES_REQUESTED** [review]

File: t/helper/test-quote.c (hunk 4 above)
Comment (garimasi514) [reply t/helper/test-quote.c]: Duh!

File: t/helper/test-quote.c [hunk 5]
```diff
@@ -0,0 +1,30 @@
+#include "test-tool.h"
+#include "quote.h"
+#include "strbuf.h"
+#include "string.h"
+
+int cmd__quote_buf_pretty(int argc, const char **argv)
+{
+	struct strbuf buf_payload = STRBUF_INIT;
+	char nullString[2];
```
Comment (jrbriggs) [inline t/helper/test-quote.c:9]: This won't actually allocate a null string, unless my C is very rusty. Maybe try `char* nullString = 0` which should expose a panic in the `sq_quote_buf_pretty` where we dereference the pointer when testing for null or empty.

**Pull request closed without merging**